controllers/database.go

```diff
@@ -37,47 +38,49 @@ var mariadbTemplates = []string{
 }
 
 // extract to var for mocking in testing
-var ConnectAndQueryDatabase = func(host, port, username, password, dbname string, dbConnectionTimeout time.Duration) bool {
+var ConnectAndQueryDatabase = func(host, port, username, password, dbname string, dbConnectionTimeout time.Duration) (bool, string) {
 	// Create a context with a timeout of 1 second
 	ctx, cancel := context.WithTimeout(context.Background(), dbConnectionTimeout)
 	defer cancel()
 
 	connectionString := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", username, password, host, port, dbname)
 	db, err := sql.Open("mysql", connectionString)
 	if err != nil {
-		return false
+		return false, err.Error()
 	}
 	defer db.Close()
 
 	testStatement := "SELECT 1;"
 	_, err = db.QueryContext(ctx, testStatement)
-	return err == nil
+	return err == nil, ""
 }
 
 func (r *DSPAReconciler) isDatabaseAccessible(ctx context.Context, dsp *dspav1alpha1.DataSciencePipelinesApplication,
-	params *DSPAParams) bool {
+	params *DSPAParams) (bool, string) {
 	log := r.Log.WithValues("namespace", dsp.Namespace).WithValues("dspa_name", dsp.Name)
 
 	if params.DatabaseHealthCheckDisabled(dsp) {
-		log.V(1).Info("Database health check disabled, assuming database is available and ready.")
-		return true
+		infoMessage := "Database health check disabled, assuming database is available and ready."
+		log.V(1).Info(infoMessage)
+		return true, infoMessage
 	}
 
 	log.Info("Performing Database Health Check")
 	databaseSpecified := dsp.Spec.Database != nil
 	usingExternalDB := params.UsingExternalDB(dsp)
 	usingMariaDB := !databaseSpecified || dsp.Spec.Database.MariaDB != nil
 	if !usingMariaDB && !usingExternalDB {
-		log.Info("Could not connect to Database: Unsupported Type")
-		return false
+		errorMessage := "Could not connect to Database: Unsupported Type"
+		log.Error(nil, errorMessage)
+		return false, errorMessage
 	}
 
 	decodePass, _ := b64.StdEncoding.DecodeString(params.DBConnection.Password)
 	dbConnectionTimeout := config.GetDurationConfigWithDefault(config.DBConnectionTimeoutConfigName, config.DefaultDBConnectionTimeout)
 
 	log.V(1).Info(fmt.Sprintf("Database Heath Check connection timeout: %s", dbConnectionTimeout))
 
-	dbHealthCheckPassed := ConnectAndQueryDatabase(params.DBConnection.Host,
+	dbHealthCheckPassed, message := ConnectAndQueryDatabase(params.DBConnection.Host,
```

Comment:
alternatively/additionally to above, we could _just_ return an `error`, and then assume success/failure below based on `err == nil`